instrumentation/apache-httpclient/apache-httpclient-5.0/src/test/java/io/opentelemetry/javaagent/instrumentation/apachehttpclient/v5_0/ConnectionTest.java

```diff
@@ -0,0 +1,203 @@
+                            .hasAttribute(AttributeKey.longKey("http.status_code"), 200L)
+                            .hasStatus(StatusData.unset())
+                    ));
+  }
+
+  @Test
+  void testGarbageResponse() {
+    final String urlPath = "/garbage";
+
+    mockServer.stubFor(get(urlPathEqualTo(urlPath))
+        .willReturn(aResponse()
+            .withFault(Fault.RANDOM_DATA_THEN_CLOSE)
+        )
+    );
+
+    Throwable thrown = catchThrowable(() -> {
+      try (CloseableHttpClient client = HttpClients.createDefault()) {
+        final ClassicHttpRequest httpGet = ClassicRequestBuilder.get(mockServer.url(urlPath))
+            .addHeader("Content-Type", "application/json").build();
+        client.execute(httpGet, response -> null);
+      }
+    });
+
+    assertThat(instrumentation.waitForTraces(1))
+        .hasSize(1)
+        .hasTracesSatisfyingExactly(
+            trace ->
+                trace
+                    .hasSize(1)
+                    .hasSpansSatisfyingExactly(
+                        span -> span
+                            .hasName("GET")
+                            .hasKind(SpanKind.CLIENT)
+                            .hasAttribute(AttributeKey.stringKey("http.method"), "GET")
+                            .hasAttribute(AttributeKey.stringArrayKey("http.request.header.content_type"),
+                                List.of("application/json"))
+                            .hasAttribute(AttributeKey.stringKey("http.request.body"), "null")
+                            .hasException(thrown)
+                            .hasStatus(StatusData.error())
+                    ));
+  }
+
+  @Test
+  void testServerError() {
+    final String urlPath = "/server-error";
+
+    mockServer.stubFor(get(urlPathEqualTo(urlPath))
+        .willReturn(aResponse()
+            .withStatus(500)
+        )
+    );
+
+    try (CloseableHttpClient client = HttpClients.createDefault()) {
+      final ClassicHttpRequest httpGet = ClassicRequestBuilder.get(mockServer.url(urlPath))
+          .addHeader("Content-Type", "application/json").build();
+      client.execute(httpGet, response -> null);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    assertThat(instrumentation.waitForTraces(1))
+        .hasSize(1)
+        .hasTracesSatisfyingExactly(
+            trace ->
+                trace
+                    .hasSize(1)
+                    .hasSpansSatisfyingExactly(
+                        span -> span
+                            .hasName("GET")
+                            .hasKind(SpanKind.CLIENT)
+                            .hasAttribute(AttributeKey.stringKey("http.method"), "GET")
+                            .hasAttribute(AttributeKey.stringArrayKey("http.request.header.content_type"),
+                                List.of("application/json"))
+                            .hasAttribute(AttributeKey.stringKey("http.request.body"), "null")
```

Comment:
To make sure we can distinguish `"null"` and `null`, please add a test showing that a response with the `null` as body (i.e., 4 charactersm content-type text/plain) results in the `"\"null\""` string.